products/platform/documentation-site/research/search-design-doc.md

```diff
@@ -1,15 +1,178 @@
+
+#### Where/when to run the scraper
+
+For the MVP, we will run the crawler in a scheduled job once every 24 hours in Circle CI.
+
+#### Private repos
+
+---
+
+### Code Location
+
+The code will live in a new repo (tentatively titled something like `docs-landing-page`).
+
+### Testing Plan
+
+#### Environments
+
+For an MVP, I'm not certain it's necessary to have different environments for the search input and the scraper.
+
+##### Testing the scraper
+
+###### New records should appear in index after a crawl
+
+1. Search for term that returns zero results
+1. Add document that contains term
+1. Run crawler
+1. Search for recently added term
+1. Assert that term is in index
+
+###### Old records should disappear from index after a crawl
+
+1. Search for term that exists in index
+1. Remove document that contains term from index
+1. Run crawler
+1. Search for recently deleted term
+1. Assert that term isn't in index
+
+###### Records should reflect edits after a crawl
+
+1. Search for record that exists in index
+1. Edit document
+1. Run crawler
+1. Search for recently edited record
+1. Assert that record reflects recent edit
+
+##### Testing the search functionality
+
+Once we have records in our Algolia index and the DocSearch frontend library on a page, we should be able to test the search functionality the same way the end user would use it.
+
+### Logging
+
+Algolia includes analytics, such as ["Most popular searches and results most often returned by the search engine."](https://www.algolia.com/products/analytics/?tab=popular)
+
+### Debugging
+
+The Docker image for the scraper is available on [Docker Hub](https://hub.docker.com/r/algolia/docsearch-scraper). Once we have a config file, we can [run the scraper locally](https://community.algolia.com/docsearch/run-your-own.html#run-the-crawl-from-the-docker-image).
+
+#### Debugging the scraper
+
+#### Debugging the search functionality
+
+We could search for the same term in both the Algolia UI and the landing page to confirm if they both return the same results.
+
+### Caveats
+
+- Access Control: By making the documentation landing page public, it becomes difficult, if not impossible, to allow an unauthenticated user to search documentation from private sources (like a private Github repo). One suggestion is that search results lead to public documentation sources, and we include links to private documentation in the public sources. That way, we can rely on existing access control solutions, while still providing a decent solution. One of the major downsides of that approach is auditing, adding, updating, and maintaining links to/from the public and private sources.
+
+### Security Concerns
+
+If the search input dispatched a request directly to the Algolia API, then there'd be minimal risk of the documentation search negatively affecting other systems. If the request was proxied through our API, then we should be able to rely on any existing DOS protections.
```

Comment:
What's the suggested implementation and how does it affect this?

Reply:
Since the request is dispatched directly to Algolia, I updated the security concerns, and removed this chunk in https://github.com/department-of-veterans-affairs/va.gov-team/pull/5248/commits/dce864810f76a161676225fca07ca3fdaa64671d.